bin/install-pr

```diff
@@ -9,63 +9,193 @@
 
 set -eo pipefail
 
+cleanup() {
+  # Remove temporary directory.
+  rm -rf "$archive_tmp"
+  
+  if [ "$installed_cli" -eq 1 ]
+  then
+    echo ""
+    echo "Linkerd CLI installed!"
+    echo "Usage:"
+    echo "    bin/linkerd $context_flag version"
+  elif [ "$installed_cli" -eq 0 ]
+  then
+    echo ""
+    echo "Linkerd CLI install failed!"
+  fi
+
+  if [ "$linkerd_check" -eq 1 ]
+  then
+    echo ""
+    echo "'linkerd check' passed!"
+    echo "Usage:"
+    echo "    bin/linkerd ${context_flag[*]} check"
+    echo "    kubectl ${context_flag[*]} -n linkerd get pods"
+  elif [ "$linkerd_check" -eq 0 ]
+  then
+    echo ""
+    echo "'linkerd check' failed!"
+    echo "Check again before debugging:"
+    echo "    bin/linkerd ${context_flag[*]} check --pre"
+    echo "    bin/linkerd ${context_flag[*]} check"
+    echo "    kubectl ${context_flag[*]} -n linkerd get pods"
+  fi
+
+  if [[ $is_kind -eq 1 && $kind_setup -eq 0 ]]
+  then
+    echo "Script failed at or before KinD cluster creation."
+  fi
+}
+
+trap cleanup EXIT
+
+# Initialize variables used throughout script.
+archive_tmp=
+context_flag=
+
+installed_cli=0
+kind_setup=0
+linkerd_check=0
+is_kind=0
+
+linkerd2_pulls_url="https://api.github.com/repos/linkerd/linkerd2/pulls"
+linkerd2_kind_integration_url="https://api.github.com/repos/linkerd/linkerd2/actions/workflows/kind_integration.yml"
+
+# Read script flags and arguments.
+while :
+do
+  case $1 in
+    -h|--help)
+      echo "Install Linkerd with the changes made in a GitHub Pull Request."
+      echo ""
+      echo "Note:"
+      echo "    If installing into an existing Kubernetes cluster, this script"
+      echo "    assumes the cluster is not a KinD cluster."
+      echo ""
+      echo "Usage:"
+      echo "    # Install Linkerd into the 'kubectl' current cluster"
+      echo "    bin/install-pr ####"
+      echo ""
+      echo "    # Install Linkerd into a new KinD cluster"
+      echo "    bin/install-pr [-k|--kind] ####"
+      installed_cli=2
+      linkerd_check=2
+      exit 0
+      ;;
+    -k|--kind)
+      is_kind=1
+      ;;
+    -?*)
+      echo "Error: Unknown option: $1" >&2
+      exit 1
+      ;;
+    *)
+      break
+  esac
+  shift
+done
+
 pr=$1
 
 if [ -z "$pr" ]
 then
-  echo "usage: ${0##*/} (PR number)" >&2
+  echo "Usage: ${0##*/} (PR number)" >&2
   exit 1
 fi
 
 if [ -z "$GITHUB_TOKEN" ]
 then
   # shellcheck disable=SC2016
-  echo 'Generate a personal access token at https://github.com/settings/tokens and set it in the $GITHUB_TOKEN env var'
+  echo 'Error: Generate a personal access token at https://github.com/settings/tokens and set it in the $GITHUB_TOKEN env var'
   exit 1
 fi
 
+# Get the URL for downloading the artifacts archive.
 auth="Authorization: token $GITHUB_TOKEN"
+branch=$(curl -sL "$linkerd2_pulls_url/$pr" | jq -r '.head.ref')
```

Comment:
Is it not necessary to add the auth header for these API requests?  Even if it's not required, it might still be a good idea?  We might get higher rate-limits, for example.

Reply:
Yep it's not necessary but I can add it back!